`pytest-xdist` detection does not work when loaded via `PYTEST_PLUGINS`

**Describe the bug**

When `pytest-xdist` plugin is loaded via `PYTEST_PLUGINS` environment variable rather than being autoloaded via `pytest`, it is registered under `xdist.plugin` name rather than `xdist` and the logic responsible for detecting it:

https://github.com/syrupy-project/syrupy/blob/086034ab091a6b8f53017a51f928f84388b0fd5e/src/syrupy/__init__.py#L230-L232

does not work correctly.

To the best of my knowledge, this is by design. Pytest plugins have no "canonical" names (`get_canonical_name()` is just an unique integer for every plugin registered in the session), and they are recorded under the names given during registration. For autodiscovery and `-p` argument, it's the entry point name (`xdist`). For `PYTEST_PLUGINS`, where you need to specify the entry point directly, it's that (`xdist.plugin`). Technically, a caller can also register it explicitly using any other name.

**To reproduce**

(in test environment for syrupy)

```
export PYTEST_DISABLE_PLUGIN_AUTOLOAD=1
export PYTEST_PLUGINS=syrupy,xdist.plugin
pytest tests/integration/test_snapshot_skipped.py::test_mark_skipped_snapshots
```

Example error:

```pytb
$ pytest tests/integration/test_snapshot_skipped.py::test_mark_skipped_snapshots 
========================================================= test session starts =========================================================
platform linux -- Python 3.14.6, pytest-9.1.1, pluggy-1.6.0
rootdir: /tmp/syrupy
configfile: pyproject.toml
plugins: syrupy-5.5.1
collected 3 items                                                                                                                     

tests/integration/test_snapshot_skipped.py ..F                                                                                  [100%]

============================================================== FAILURES ===============================================================
_______________________________________________ test_mark_skipped_snapshots[xdist_two] ________________________________________________

run_testcases = (<Testdir local('/tmp/pytest-of-mgorny/pytest-17/test_mark_skipped_snapshots2')>, {'used': "\n            def test_use...'not-skipped': "\n            def test_skipped(snapshot):\n                assert snapshot == 'unused'\n            "})
plugin_args = ['--numprocesses', '2']

    def test_mark_skipped_snapshots(run_testcases, plugin_args):
        testdir, testcases = run_testcases
        pyfile_content = "\n\n".join([testcases["used"], testcases["mark-skipped"]])
        testdir.makepyfile(test_file=pyfile_content)
    
        result = testdir.runpytest("-v", *plugin_args)
>       result.stdout.re_match_lines(r"1 snapshot passed\.$")
E       Failed:  nomatch: '1 snapshot passed\\.$'
E            and: '============================= test session starts =============================='
E            and: 'platform linux -- Python 3.14.6, pytest-9.1.1, pluggy-1.6.0 -- /tmp/syrupy/.venv/bin/python3'
E            and: 'cachedir: .pytest_cache'
E            and: 'rootdir: /tmp/pytest-of-mgorny/pytest-17/test_mark_skipped_snapshots2'
E            and: 'created: 2/2 workers'
E            and: '2 workers [2 items]'
E            and: ''
E            and: 'scheduling tests via LoadScheduling'
E            and: ''
E            and: 'test_file.py::test_skipped '
E            and: 'test_file.py::test_used '
E            and: '[gw1] [ 50%] SKIPPED test_file.py::test_skipped '
E            and: '[gw0] [100%] PASSED test_file.py::test_used '
E            and: ''
E            and: '========================= 1 passed, 1 skipped in 0.32s ========================='
E       remains unmatched: '1 snapshot passed\\.$'

/tmp/syrupy/tests/integration/test_snapshot_skipped.py:53: Failed
-------------------------------------------------------- Captured stdout setup --------------------------------------------------------
============================= test session starts ==============================
platform linux -- Python 3.14.6, pytest-9.1.1, pluggy-1.6.0 -- /tmp/syrupy/.venv/bin/python3
cachedir: .pytest_cache
rootdir: /tmp/pytest-of-mgorny/pytest-17/test_mark_skipped_snapshots2
collecting ... collected 2 items

test_file.py::test_used PASSED                                           [ 50%]
test_file.py::test_skipped PASSED                                        [100%]

--------------------------- snapshot report summary ----------------------------
2 snapshots generated.
============================== 2 passed in 0.01s ===============================
-------------------------------------------------------- Captured stdout call ---------------------------------------------------------
============================= test session starts ==============================
platform linux -- Python 3.14.6, pytest-9.1.1, pluggy-1.6.0 -- /tmp/syrupy/.venv/bin/python3
cachedir: .pytest_cache
rootdir: /tmp/pytest-of-mgorny/pytest-17/test_mark_skipped_snapshots2
created: 2/2 workers
2 workers [2 items]

scheduling tests via LoadScheduling

test_file.py::test_skipped 
test_file.py::test_used 
[gw1] [ 50%] SKIPPED test_file.py::test_skipped 
[gw0] [100%] PASSED test_file.py::test_used 

========================= 1 passed, 1 skipped in 0.32s =========================
======================================================= short test summary info =======================================================
FAILED tests/integration/test_snapshot_skipped.py::test_mark_skipped_snapshots[xdist_two] - Failed:  nomatch: '1 snapshot passed\\.$'
===================================================== 1 failed, 2 passed in 0.53s =====================================================
```

**Expected behavior**

Tests passing.

**Screenshots**

N/A

**Environment (please complete the following information):**

- OS: Gentoo Linux amd64
- Syrupy Version: 086034ab091a6b8f53017a51f928f84388b0fd5e
- Python Version: 3.14.6

**Additional context**

I'll submit a PR shortly.

--- a/src/syrupy/__init__.py
+++ b/src/syrupy/__init__.py
@@ -228,7 +228,9 @@ class DeferXDist:
 
 
 def pytest_configure(config: pytest.Config) -> None:
-    if config.pluginmanager.hasplugin("xdist"):
+    if config.pluginmanager.hasplugin("xdist") or config.pluginmanager.hasplugin(
+        "xdist.plugin"
+    ):
         config.pluginmanager.register(DeferXDist())
 
 

--- a/tests/integration/test_snapshot_xdist.py
+++ b/tests/integration/test_snapshot_xdist.py
@@ -78,3 +78,37 @@ def test_without_xdist_installed(testdir):
     assert result.ret == 0
     result.stdout.no_fnmatch_line("*PluginValidationError*")
     result.stdout.no_fnmatch_line("*unknown hook*")
+
+
+def test_p_xdist(testdir, monkeypatch):
+    """Test detecting pytest-xdist registered via "-p xdist"."""
+    testdir.makepyfile(
+        """
+        def test_a(snapshot):
+            assert 1 == snapshot
+        """
+    )
+
+    monkeypatch.setenv("PYTEST_DISABLE_PLUGIN_AUTOLOAD", "1")
+    monkeypatch.setenv("PYTEST_PLUGINS", "syrupy")
+    result = testdir.runpytest(
+        "-v", "-p", "xdist", "--numprocesses", "2", "--snapshot-update"
+    )
+    result.stdout.re_match_lines((r"1 snapshot generated",))
+    assert result.ret == 0
+
+
+def test_pytest_plugins_xdist(testdir, monkeypatch):
+    """Test detecting pytest-xdist registered via PYTEST_PLUGINS."""
+    testdir.makepyfile(
+        """
+        def test_a(snapshot):
+            assert 1 == snapshot
+        """
+    )
+
+    monkeypatch.setenv("PYTEST_DISABLE_PLUGIN_AUTOLOAD", "1")
+    monkeypatch.setenv("PYTEST_PLUGINS", "syrupy,xdist.plugin")
+    result = testdir.runpytest("-v", "--numprocesses", "2", "--snapshot-update")
+    result.stdout.re_match_lines((r"1 snapshot generated",))
+    assert result.ret == 0

